crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -36,26 +36,32 @@ def restore
 
       protected
 
-      def knife
-        [:nodes, :roles, :clients, :databags].each do |type|
-          Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
-            file = Pathname.new(file)
-            next unless file.extname == ".json"
-            record = JSON.load(file.read)
-            if file.basename.to_s.include?("-default.json")
-              bc_name = file.basename.to_s.split("-").first
-              Proposal.create(barclamp: bc_name, name: "default", properties: json)
-              SchemaMigration.run_for_bc(bc_name)
-            else
-              record.save
+      def restore_chef
+        begin
+          [:nodes, :roles, :clients, :databags].each do |type|
+            Dir.glob(@data.join("knife", type.to_s, "**", "*")).each do |file|
+              file = Pathname.new(file)
+              next unless file.extname == ".json"
+              record = JSON.load(file.read)
+              if file.basename.to_s.include?("-default.json") && type == :databags
+                bc_name = file.basename.to_s.split("-").first
+                Proposal.create(barclamp: bc_name, name: "default", properties: record)
+                SchemaMigration.run_for_bc(bc_name)
+              else
+                record.save
+              end
             end
           end
+        rescue Errno::ECONNREFUSED => e
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `e`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)